beluga/test/beluga/algorithm/test_estimation.cpp

```diff
@@ -253,4 +268,84 @@ TEST_F(ScalarEstimation, NonUniformWeightOverload) {
   ASSERT_NEAR(standard_deviation, 2.026, kTolerance);
 }
 
+TEST_F(PoseCovarianceEstimation, MultiVariateNormalSE3) {
+  constexpr double kTolerance = 0.01;
+  const auto expected_mean =
+      Sophus::SE3d{Sophus::SO3d::exp(Eigen::Vector3d{-0.17, 0.25, 0.1}), Eigen::Vector3d{1.0, 2.0, 3.0}};
+  const Eigen::Matrix<double, 6, 6> expected_cov = Eigen::Matrix<double, 6, 6>::Identity() * 2e-1;
+  auto distribution = beluga::MultivariateNormalDistribution{expected_mean, expected_cov};
+  const auto samples = beluga::views::sample(distribution) |   //
+                       ranges::views::take_exactly(500'000) |  //
```

Comment:
@serraramiro1 consider using fewer samples, `100'000` should give you a good enough approximation and will improve test runtime for debug builds.